oauth/lib/oauth2.cpp

```diff
@@ -0,0 +1,987 @@
+  set this property to an empty list value, to avoid accidentally ignoring
+  future errors.
+
+  This property is difficult to use properly from a QML context, as the
+  \c{QSslError} type is opaque to QML; as such, this property is mostly
+  useful for C++ clients.
+
+  It is an optional property.
+
+  \sa sslErrors()
+*/
+QList<QSslError> SailfishOS::OAuth::OAuth2::ignorableSslErrors() const
+{
+    Q_D(const OAuth2);
+    return d->m_ignorableSslErrors;
+}
+
+void SailfishOS::OAuth::OAuth2::setIgnorableSslErrors(const QList<QSslError> &errors)
+{
+    Q_D(OAuth2);
+    BASIC_Q_PROPERTY_SETTER(d->m_ignorableSslErrors, errors, ignorableSslErrorsChanged);
+}
+
+/*!
+  \property SailfishOS::OAuth::OAuth2::error
+  \brief This property holds the error code associated with the most recent
+         error encountered by the helper.
+
+  The possible values of the code are described
+  \l{SailfishOS::OAuth::OAuth2::OAuth2Error}{here}.
+
+  In general, if any error (other than \c{NoError}) occurs, then
+  the flow has failed for some reason.
+*/
+SailfishOS::OAuth::OAuth2::OAuth2Error SailfishOS::OAuth::OAuth2::error() const
+{
+    Q_D(const OAuth2);
+    return d->m_error;
+}
+
+/*!
+  \property SailfishOS::OAuth::OAuth2::errorMessage
+  \brief This property holds the message associated with the most recent error
+         encountered by the helper.
+
+  The message is not translated or user-friendly, but is intended for developer
+  consumption only.
+
+  It may contain sensitive information, and as such it should not be logged
+  persistently.
+*/
+QString SailfishOS::OAuth::OAuth2::errorMessage() const
+{
+    Q_D(const OAuth2);
+    return d->m_errorMessage;
+}
+
+/*!
+  \brief Parse the given \a redirectUri and construct a map of key/value pairs.
+
+  Both the URI fragment and the query parameters will be parsed, and the
+  key/value pairs will be inserted into the returned dictionary.
+
+  The client can then look for the existence of appropriate entries
+  (e.g. "error", "code", "access_token", etc.
+
+  Any errors encountered during parsing are assumed to be non-fatal,
+  and so while the \l{errorMessage} will be set, the \l{error} will
+  not be.  The client must determine whether the flow has failed,
+  for example by determining whether the output contains the
+  appropriate token.
+*/
+QVariantMap SailfishOS::OAuth::OAuth2::parseRedirectUri(const QString &redirectUri)
+{
+    Q_D(OAuth2);
+    QString errorMessage;
+    const QVariantMap data = SailfishOS::OAuth::parseRedirectUri(redirectUri, &errorMessage);
+    if (!errorMessage.isEmpty()) {
+        // don't emit errorChanged, as the error may not be fatal.
+        d->m_errorMessage = errorMessage;
+        emit errorMessageChanged();
+    }
+    return data;
+}
+
+/*!
+  \brief Joins the specified \a scopes with the specified \a separator and returns the result as a valid \c{scope} string
+*/
+QString SailfishOS::OAuth::OAuth2::generateScope(
```

Comment:
Maybe you can have setScope overload that you can pass QStringList and separator (with default value)

Reply:
I can add such overload also.  The current method exists to make it simpler to use from QML (see e.g. how the OAuth2AcPkce type offers a "scopes" property and then binds the "scope" property to it through a generateScope() filter).

Reply:
done